src/metrics-logger.ts

```diff
@@ -0,0 +1,49 @@
+'use strict';
+import { Request } from './custom-typings';
+import { Route, Response, Server } from 'restify';
+import { StatsD, StatsDOptions } from './statsd';
+
+type HighResolutionTime = [number, number]; // [seconds, nanoseconds]
+
+export const DEFAULT_KEY_NAME: string = 'handler-0';
+
+export function register(server: Server, statsDOptions: StatsDOptions) {
+    let statsD: StatsD = new StatsD(statsDOptions);
+    let logger = new LogFactory(statsD).createLogger();
+
+    server.on('after', (request: Request, response: Response, route: Route, error) => {
+        logger(request, route);
+    });
+}
+
+export class LogFactory {
```

Comment:
i'm concerned about the name of this class, as it has no relation to logging and could cause confusion. Maybe something more related to the purpose... `AfterRouteHandlerFactory` ? i'm not sure though.